### PR TITLE
New ebuild for logfury

### PR DESCRIPTION
**Pull request by WPettersson**
Logfury is a boilerplate method-call logging library for python

### CODE REVIEW

File: dev-python/logfury/logfury-0.1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Boilerplate library for logging method calls"
+HOMEPAGE="https://github.com/ppolewicz/logfury https://pypi.org/project/logfury/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
```
Comment (gyakovlev) [inline dev-python/logfury/logfury-0.1.2.ebuild]: I see `doc` useflag, but it's not used in any way)
Same goes for test, tests are not run.

**Comment by WPettersson** [comment]
Updated commit message, and removed the unused USE flags. Both done by amending my original commit, I think that's the right way to go about this to keep history clean?

**Comment by gyakovlev** [comment]
Yes, keeping history clean is important, you did it correctly, thanks!

I'll test it a bit later, otherwise looks good to me.

File: dev-python/logfury/logfury-0.1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	virtual/python-funcsigs[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}] "
```
Comment (gyakovlev) [inline dev-python/logfury/logfury-0.1.2.ebuild]: non-conditional deps should be sorted  alphabetically.
Also remove the whitespace before the closing `"`

Also, RDEPEND is missing.
DEPEND != RDEPEND by default if EAPI=6
https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-650007.3.3

you should probably add six and funcsigs to RDEPEND,
and add `${RDEPEND}` to `DEPEND` along with setuptools.

check out for example dev-python/PyGithub/PyGithub-1.40_alpha4.ebuild how to do that. I just picked random ebuild that can hint you.

File: dev-python/logfury/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="project">
```
Comment (gyakovlev) [inline dev-python/logfury/metadata.xml]: you are not a project =) change to `type="person"`

File: dev-python/logfury/logfury-0.1.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit distutils-r1
+
+DESCRIPTION="Boilerplate library for logging method calls"
+HOMEPAGE="https://github.com/ppolewicz/logfury https://pypi.org/project/logfury/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
```
Comment (gyakovlev) [inline dev-python/logfury/logfury-0.1.2.ebuild]: Nope, it's `BSD`.

project github says it's BSD 3-clause, 
https://github.com/ppolewicz/logfury/blob/master/LICENSE.txt
in gentoo it's called just `BSD`.

If you look into that file in licenses/BSD, it contains exactly 3 clauses =)

**Comment by WPettersson** [comment]
Thanks for being patient with me. I've set myself as a person, not a project, fixed the license and fixed the DEPENDS/RDEPENDS. I'm not 100% sure on the whitespace, I couldn't find any style guidelines for ebuilds so I've tried to copy other ebuilds which means there is no empty line between the end of RDEPEND and the start of DEPEND.

**Comment by gyakovlev** [comment]
Hey, no problem at all. That's a learning experience and you are doing great!

Here is the space I was referring to:
```
dev-python/six[${PYTHON_USEDEP}] "
                                ^ see that sucker before the final " ? =)
```



There should be an empty line between RDEPEND and DEPEND.

So last nit, add an empty line RDEPEND block, so there is a line before DEPEND starts and I'm going to merge it =)

**Comment by WPettersson** [comment]
Ok, that should be it then.

**Comment by gyakovlev** [comment]
Thanks!
I'll wait until ci bot comments and will merge it.

**Pull request closed without merging**